src/Analysis/Ast/Impl/Types/PythonType.cs

```diff
@@ -23,8 +23,7 @@
 namespace Microsoft.Python.Analysis.Types {
     [DebuggerDisplay("{Name}")]
     internal class PythonType : LocatedMember, IPythonType {//, IEquatable<IPythonType> {
-        private readonly object _lock = new object();
-        private readonly string _name;
+        protected readonly object _lock = new object();
```

Comment:
No protected variables please. If something is protected it should be readonly property. Why do we need protected lock?

Reply:
This was because I had to use the same lock in PythonClassType.

Reply:
Does it have to be the same lock?

Reply:
I was concerned about reading and writing members so I wanted to use the same lock, but after looking at it more it's really about pushing and popping with two different threads, not writing and reading.